src/codegen_types/datetime.cr

```diff
@@ -1,11 +1,11 @@
 module AST
   class DateTimePrimitiveType
     def typescript_decode(expr)
-      "moment.utc(#{expr}, \"YYYY-MM-DDTHH:mm:ss.SSS\").toDate()"
+      "new Date(#{expr})"
     end
 
     def typescript_encode(expr)
-      "moment.utc(#{expr}).format(\"YYYY-MM-DDTHH:mm:ss.SSS\")"
+      "#{expr}.toISOString()"
```

Comment:
toISOString() addes "Z" on the end. Current format has no timezone information. Also, the date must first be converted to UTC. This is not a problem for the server (which localtime = utc), but for the targets web and nodeclient this is an issue.

When decoding it needs to be converted from UTC to localtime, as well.

Reply:
toISOString automatically converts to UTC timezone (as [stated on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/UTC)), so the only adjustment to be made would be removing the "Z" from the string. However, changing a Date from UTC to local timezone is a headache of its own.

Reply:
Question: if the Date object is aware that its timezone is different from the local timezone and getHours(), getMonth(), etc, return their local values, when does a UTC Date object have unexpected behavior?

Reply:
We have tested UTC Date behavior when used with moment and it works smoothly, so that shouldn't be an issue. That said, when decoding, "Z" must be appended to the end of the ISO string, so that the Date constructor can be aware of its UTC timezone.